src/metaopt/io/database/__init__.py

```diff
@@ -148,4 +148,70 @@ class DatabaseError(RuntimeError):
     pass
 
 
+# Include current implementations of the wrapper
 from metaopt.io.database.mongodb import MongoDB  # noqa
+
+
+class DatabaseType(SingletonType):
+    """Instantiate appropriate wrapper for the infrastructure based on input
+    argument, ``dbtype``.
+
+    Attributes
+    ----------
+    all : list of `AbstractDB` classes
+       Updated to contain `all` possible implementations currently. Check out code.
+    typenames : list of str
+       Names of implemented wrapper classes, correspond to possible ``dbtype``
+       values.
+
+    """
+
+    all = [AbstractDB, MongoDB]
+    typenames = list(map(lambda x: x.__name__, all))
+
+    def __call__(cls, dbtype='AbstractDB', *args, **kwargs):
```

Comment:
Shouldn't it be `dbtype=None`? I don't think we want AbstractDB to be ever instantiated.

Reply:
It cannot be, eitherways 😛

Reply:
So why AbstractDB by default if it can't be? None seems less confusing to me, but I might miss something.

Reply:
`None` raises an exception somewhere 'random'. In particular when doing `dbtype.lower()`. Passing 'AbstractDB' raises a more natural and informative error.